src/core/ddsi/src/ddsi_wraddrset.c

```diff
@@ -347,14 +347,17 @@ static readercount_cost_t calc_locator_cost (const struct cover *c, int lidx, bo
   if (rdidx == c->nreaders)
     goto no_readers;
 
+#ifdef DDS_HAS_SHM
```

Comment:
Is there really a need to do `#ifdef DDS_HAS_SHM` here? I had tried to avoid those as much as I could, so it is a bit of a shame if it needs to be re-added. It perhaps would make more sense to define `NN_LOCATOR_KIND_SHEM` unconditionally in `q_protocol.h`.

Reply:
At the moment I put this inside the ifdefs because NN_LOCATOR_KIND_SHEM is only defined when this compile flag is set.

I was more thinking in the direction of a more generic checking function which compares the locator kind together with the mask and makes a decision there.

Reply:
Good point, it would probably be vastly more sensible to store the locator type directly in an accessible place (we have `struct locset` after all) and simply check whether the kind is included/not excluded.

Or, perhaps even wiser, bite the bullet and check the locator kinds in all places where it deals with locators as stored in the readers' address sets — that is, `wras_flatten_locs_prealloc` and expanding `copy_addrset_into_addrset` in `wras_collect_all_locs` and filtering right there (or modify `wras_calc_locators`).